packages/office-ui-fabric-react/src/components/Announced/examples/Announced.QuickActions.Example.tsx

```diff
@@ -0,0 +1,289 @@
+          </Dialog>
+        </MarqueeSelection>
+      </>
+    );
+  }
+
+  public componentWillUnmount() {
+    if (this.state.showItemIndexInView) {
+      const itemIndexInView = this._detailsList!.current!.getStartItemIndexInView();
+      alert('unmounting, getting first item index that was in view: ' + itemIndexInView);
+    }
+  }
+
+  private _onRenderRow(props: IDetailsRowProps): JSX.Element {
+    return <DetailsRow {...props} />;
+  }
+
+  private _onRenderItemColumn(item: any, index: number, column: IColumn) {
+    const fieldContent = item[column.fieldName || ''];
+
+    switch (column.key) {
+      case 'column1':
+        return (
+          <div>
+            {fieldContent}
+            <IconButton
+              iconProps={{ iconName: 'MoreVertical' }}
+              role="button"
+              aria-haspopup={true}
+              aria-label="Show actions"
+              onRenderMenuIcon={nullFunction}
+              styles={{ root: { float: 'right', height: 'inherit' } }}
+              menuProps={{
+                items: [
+                  {
+                    key: 'delete',
+                    text: 'Delete',
+                    onClick: () => this._deleteItem(index)
+                  },
+                  {
+                    key: 'rename',
+                    text: 'Rename',
+                    onClick: () => this._renameItem(item, index)
+                  }
+                ]
+              }}
+            />
+          </div>
+        );
+
+      default:
+        return <span>{fieldContent}</span>;
+    }
+  }
+
+  private _renderAnnounced(): JSX.Element | undefined {
+    const { announced } = this.state;
+    return announced;
+  }
+
+  private _deleteItem(index: number): void {
+    const items = this.state.items;
+    items.splice(items.indexOf(items[index]), 1);
+    for (let i = 0; i < items.length; i++) {
+      console.log(i + ': ' + items[i].name);
+    }
+    this.setState(
+      {
+        items: items,
+        announced: <Announced message={`Item deleted`} />
+      },
+      () => {
+        if (this._detailsList.current) {
+          this._detailsList.current.forceUpdate();
+        }
+      }
+    );
+    return;
+  }
+
+  private _renameItem(item: any, index: number): void {
+    this.setState({
+      renameDialogOpen: true,
+      dialogContent: (
+        <>
+          <TextField componentRef={this._textField} label="Rename" value={item.name} />
+          <DialogFooter>
+            <PrimaryButton onClick={this._updateItemName.bind(this, item, index)} text="Save" />
+          </DialogFooter>
+        </>
+      )
+    });
+    return;
+  }
+
+  private _updateItemName(item: any, index: number): void {
+    if (this._textField && this._textField.current) {
+      const items = this.state.items;
+      items[index].name = this._textField.current.value || items[index].name;
+      this.setState(
+        {
+          renameDialogOpen: false,
+          items: items,
```

Comment:
I wonder if you did `items: [ ...items ]` here that the `DetailsList` would update removing the need for the `forceUpdate()` call below. The reason I think it _might_ work is because the `List` does reference checking for equality which replacing the entire items array should trigger.

Reply:
Not a blocker, but if we can remove explicit calls to `forceUpdate` and prefer other methods across code that would be ideal. It might not be possible here tho, I haven't tried!

Reply:
That worked!

Reply:
Awesome 🙇